libsolidity/formal/SMTEncoder.cpp

```diff
@@ -348,13 +348,46 @@ void SMTEncoder::endVisit(VariableDeclarationStatement const& _varDecl)
 
 }
 
+bool SMTEncoder::visit(Assignment const& _assignment)
+{
+	auto const& left = _assignment.leftHandSide();
+	auto const& right = _assignment.rightHandSide();
+
+	if (auto const* memberAccess = isEmptyPush(left))
+	{
+		right.accept(*this);
+		left.accept(*this);
+
+		auto const& memberExpr = memberAccess->expression();
+		auto& symbArray = dynamic_cast<smt::SymbolicArrayVariable&>(*m_context.expression(memberExpr));
+		smtutil::Expression oldElements = symbArray.elements();
+		smtutil::Expression oldLength = symbArray.length();
+		symbArray.increaseIndex();
+		m_context.addAssertion(symbArray.elements() == smtutil::Expression::store(
+			oldElements,
+			oldLength - 1,
+			expr(right)
+		));
```

Comment:
Can you explain me here, why `oldLength -1`, I am not exactly sure about it?

Reply:
Because the actual `push` was already handled inside `left.accept`, so the length was already increased. Here we only assign the right hand side to the newly created last position. Maybe the name `oldLength` is misleading, since the length doesn't change here.

Reply:
I was thinking that it is because it length was already increased. Can you, please, than change the name of `oldLength` to `lenght` or something similar?

Reply:
Yep, will change